docs/source/overview/cli.md

```diff
@@ -0,0 +1,499 @@
+    max_steps: 960000,
+    report_stats_every_n_steps: 32000,
+    save_checkpoint_every_n_steps: 1000,
+    n_validation_samples: 64000,
+    n_test_samples: 960000,
+    cpu_n_threads: 32,
+    use_cuda: True,
+    data_parallel: True,
+    logging_verbosity: 2,
+    checkpoint_resume: False
+}
+```
+
+#### Required parameters
+- `batch_size`:  Number of samples in one forward/backward pass (a single step).
+- `max_steps`: Total number of steps for which to train the model. 
+- `report_stats_every_n_steps`: The frequency with which to report summary statistics. 
+
+#### Optional parameters
+- `save_checkpoint_every_n_steps`: Default is 1000. The number of steps before Selene saves a new checkpoint model weights file. If this parameter is set to `None`, we will set it to the same value as `report_stats_every_n_steps`.
+- `n_validation_samples`: Default is `None`. Specify the number of validation samples in the validation set. If `None`
+   - and the data sampler you use is of type `selene_sdk.samplers.OnlineSampler`, we will by default retrieve 32000 validation samples.
+   - and you are using a `selene_sdk.samplers.MultiFileSampler`, we will use all the validation samples available in the appropriate data file.
+- `n_test_samples`: Default is `None`. Specify the number of test samples in the test set. If `None`
+    - and the sampler you specified has no test partition, it is assumed that you will not be evaluating your trained model using the `evaluate` method available in `selene_sdk.TrainModel`. (i.e. `evaluate` should not be one of the operations in the `ops` list)
+    - and the sampler you use is of type `selene_sdk.samplers.OnlineSampler` (and the test partition exists), we will retrieve 640000 test samples.
+    - and the sampler you use is of type `selene_sdk.samplers.MultiFileSampler`, we will use all the test samples available in the appropriate data file. 
+- `cpu_n_threads`: Default is 1. The number of OpenMP threads used for parallelizing CPU operations in PyTorch.
+- `use_cuda`: Default is False. Specify whether CUDA GPUs are available for torch to use during training.  
+- `data_parallel`: Default is False. Specify whether multiple GPUs are available for torch to use during training.
+- `logging_verbosity`: Default is 2. Possible values are `{0, 1, 2}
+`. Sets the logging verbosity level:
+  - 0: only warnings are logged
+  - 1: information and warnings are logged
+  - 2: debug messages, information, and warnings are all logged
+ - `checkpoint_resume`: Default is `None`. If not `None`, you should pass in the path to a model weights file generated by `torch.save` (and can now be read by `torch.load`) to resume training. 
+ 
+#### Additional notes
+An important thing to observe about the contents of `train_model`: the [documentation for the `TrainModel` class](http://selene.flatironinstitute.org/selene.html#trainmodel) in `selene_sdk` shows that we are missing a number of arguments needed to instantiate the class.
```

Comment:
Might need to rephrase this. For example, 'attentive readers might have noticed that in the documentation there are more arguments that are required to instantiate the class. This is because they are assumed to be carried through/retrieved from other parts to ensure consistency...'